### PR TITLE
World: change spawnpoint type to int

### PR DESCRIPTION
**Pull request by tigerw**
As Vanilla does.

### CODE REVIEW

File: src/World.cpp [hunk 1]
```diff
@@ -270,9 +270,9 @@ cWorld::cWorld(
 	if (m_IsSpawnExplicitlySet)
 	{
 		LOGD("Spawnpoint explicitly set!");
-		m_SpawnX = IniFile.GetValueF("SpawnPosition", "X", m_SpawnX);
-		m_SpawnY = IniFile.GetValueF("SpawnPosition", "Y", m_SpawnY);
-		m_SpawnZ = IniFile.GetValueF("SpawnPosition", "Z", m_SpawnZ);
+		m_SpawnX = IniFile.GetValueI("SpawnPosition", "X", m_SpawnX);
+		m_SpawnY = IniFile.GetValueI("SpawnPosition", "Y", m_SpawnY);
+		m_SpawnZ = IniFile.GetValueI("SpawnPosition", "Z", m_SpawnZ);
```
Comment (bearbin) [inline src/World.cpp:275]: How compatible is this with existing decimal spawn points - will the decimal just be ignored or does this fail?

Comment (tigerw) [reply src/World.cpp:275]: It uses `atoi` to convert, and quite luckily its behaviour of "ignore everything after the number" works in our favour!

**Review by bearbin — APPROVED** [review]

**Comment by bearbin** [comment]
(well, once the clang fail is fixed)

**Comment by tigerw** [comment]
Thank you!

**Pull request merged**